### PR TITLE
Fix 'get_transactions' response.

### PR DESCRIPTION
**Pull request by cobward**
## Motivation

The response to the `get_transactions` method on the etherscan client has a number of issues:
1. `methodId` and `functionName` fields are missing.
2. A number of fields are of the type `GenesisOption`, but from my investigations only `hash` and `from` can have this format.
3. `input` does not need to be wrapped in an option. From my investigation, the response from etherscan is always `"0x"` for empty input, and the serde implementation for the `Bytes` type will process this as an empty byte array.

## Solution

1. Add the missing fields.
2. Replace `GenesisOption` with `Option` on fields that cannot start with `GENESIS`.
3. Remove the option wrapper for the `bytes` field.

## PR Checklist

- [x] Added Tests (no tests needed as the existing tests are sufficient)
- [x] Added Documentation
- [x] Updated the changelog

### CODE REVIEW

File: ethers-etherscan/src/account.rs [hunk 1]
```diff
@@ -63,7 +63,82 @@ mod jsonstring {
...
+    where
+        T: DeserializeOwned,
+        D: Deserializer<'de>,
+    {
+        let json = Cow::<'de, str>::deserialize(deserializer)?;
+        if json.is_empty() {
+            Ok(Option::None)
+        } else {
+            let value =
+                serde_json::from_str(&format!("\"{}\"", &json)).map_err(D::Error::custom)?;
```
Comment (mattsse) [inline ethers-etherscan/src/account.rs:96]: I believe this could be a one-liner by adding `.map(Some)`

File: ethers-etherscan/src/account.rs [hunk 2]
```diff
@@ -63,7 +63,82 @@ mod jsonstring {
...
+    where
+        T: DeserializeOwned,
+        D: Deserializer<'de>,
+    {
+        let json = Cow::<'de, str>::deserialize(deserializer)?;
+        if json.is_empty() || json == "0x" {
+            Ok(Option::None)
+        } else {
+            let value =
+                serde_json::from_str(&format!("\"{}\"", &json)).map_err(D::Error::custom)?;
```
Comment (mattsse) [inline ethers-etherscan/src/account.rs:132]: same here

**Review by mattsse — APPROVED** [review]
the etherscan api continues to be a bit tricky -.-

thanks for this, lgtm, smol suggestion

File: ethers-etherscan/src/account.rs (hunk 1 above)
Comment (cobward) [reply ethers-etherscan/src/account.rs]: Done

File: ethers-etherscan/src/account.rs (hunk 2 above)
Comment (cobward) [reply ethers-etherscan/src/account.rs]: and done!

File: ethers-etherscan/src/account.rs [hunk 3]
```diff
@@ -135,7 +135,10 @@ mod hex_string {
     }
 }
 
-/// Possible values for some field responses
+/// Possible values for some field responses.
+///
+/// Transactions from the Genesis block may contain fields that do not conform to the expected
+/// types.
```
Comment (kyruzic) [inline ethers-etherscan/src/account.rs:141]: Thank you for adding this explanation :)

**Pull request merged**